cli/src/commands/Test.py

```diff
@@ -40,8 +43,29 @@ def test(self):
         if not os.path.exists(spec_output):
             os.makedirs(spec_output)
 
+        selected_test_groups = self.included_groups
+
+        # exclude test groups
+        if self.excluded_groups:
+            included_groups = self.included_groups
+            if 'all' in included_groups:
+                # get available test groups
+                inventory_groups = load_inventory(path_to_inventory).list_groups()
+                effective_inventory_groups = inventory_groups + ['common']
+                included_groups = [group for group in self.test_groups if group in effective_inventory_groups]
+
+            selected_test_groups = [group for group in included_groups if group not in self.excluded_groups]
+
         # run the spec tests
-        spec_command = SpecCommand()
-        spec_command.run(spec_output, path_to_inventory, admin_user.name, admin_user.key_path, self.group)
+        if selected_test_groups:
+            spec_command = SpecCommand()
+            if 'all' in selected_test_groups:
+                selected_test_groups = ['all']
+            else:
+                self.logger.info(f'Selected test groups: {", ".join(selected_test_groups)}')
+
+            spec_command.run(spec_output, path_to_inventory, admin_user.name, admin_user.key_path, selected_test_groups)
+        else:
+            raise Exception('No test group to test')
```

Comment:
`No test group specified to run`  maybe? As your not testing a test group but running a test group to test.